cpu/cpustress.py

```diff
@@ -0,0 +1,307 @@
+            eval(run_test)
+            msg = self.__error_check()
+            self.log.info('Test: %s. ERROR Message: %s', run_test, msg)
+            self.log.info("\nEND: %s\n", method)
+
+    def cpu_serial_off_on(self, iteration, timeout):
+        """
+        Offline all the cpus serially and online again
+        offline 0 -> 99
+        online 99 -> 0
+        offline 99 -> 0
+        online 0 -> 99
+        """
+        for _ in range(iteration):
+            self.log.info("OFF-ON Serial Test %s", totalcpus)
+            for cpus in range(1, totalcpus):
+                self.log.info("cpu%s going offline" % cpus)
+                cpu.offline(cpus)
+                time.sleep(timeout)
+            self.log.info("Online CPU's in reverse order %s", totalcpus)
+            for cpus in range(totalcpus, -1, -1):
+                self.log.info("cpu%s going online" % cpus)
+                cpu.online(cpus)
+                time.sleep(timeout)
+            self.log.info("Offline CPU's in reverse order %s", totalcpus)
+            for cpus in range(totalcpus, -1, -2):
+                self.log.info("cpu%s going offline" % cpus)
+                cpu.offline(cpus)
+                time.sleep(timeout)
+            self.log.info("Online CPU's in serial")
+            for cpus in range(0, totalcpus):
+                self.log.info("cpu%s going online" % cpus)
+                cpu.online(cpus)
+                time.sleep(timeout)
+
+    def single_cpu_toggle(self, iteration, timeout):
+        """
+        Offline-online single cpu for given iteration
+        and loop over all cpus.
+        @BUG: https://lkml.org/lkml/2017/6/12/212
+        """
+        for cpus in range(1, totalcpus):
+            for _ in range(iteration):
+                self.log.info("cpu%s going offline" % cpus)
+                cpu.offline(cpus)
+                time.sleep(timeout)
+                self.log.info("cpu%s going online" % cpus)
+                cpu.online(cpus)
+
+    def cpu_toggle_one_by_one(self, iteration, timeout):
+        """
+        Wait for the given timeout between Off/On single cpu.
+        loop over all cpus for given iteration.
+        """
+        for _ in range(iteration):
+            for cpus in range(totalcpus):
+                self.log.info("cpu%s going offline" % cpus)
+                cpu.offline(cpus)
+                time.sleep(timeout)
+                self.log.info("cpu%s going online" % cpus)
+                cpu.online(cpus)
+
+    def multiple_cpus_toggle(self, iteration, timeout):
+        """
+        offline/online multiple CPUS at once
+        enable different smt options to offline multiple cpus
+        @BUG : https://lkml.org/lkml/2017/7/3/21
+        """
+        if self.__isSMT():
+            for _ in range(iteration):
+                self.log.info("SMT toggle")
+                process.run("ppc64_cpu --smt=off && ppc64_cpu --smt=on",
+                            shell=True)
+            time.sleep(timeout)
```

Comment:
i think sleep is not required as 

`ppc64_cpu --smt=off && ppc64_cpu --smt=on` run in single time i think no need to wait